src/internet-computer-programming-concepts/basic-memory-persistence/stable-variables.md

```diff
@@ -34,27 +37,58 @@ Our actor has a mutable variable `count` that is declared `stable`. It's initial
+### Stable Records with Mutable Fields
+Immutable or mutable variables of [*records with mutable fields*](/common-programming-concepts/types/records.html) could be declared stable:
 ```motoko
-{{#include _mo/stable-variables3.mo:a}}
+{{#include _mo/stable-var1.mo:b}}
 ```
 
-We define a type of an object with a public field that is mutable. We assign an instance of that object to an immutable variable named `myObject`. We could now mutate the state of our public variable.
+*Immutable variable* `r1` and *mutable variable* `r2` can be stable because they have a value of a *record with a mutable field*. 
+
+### Stable Objects with Mutable Variables
+Immutable or mutable variables of [*objects with (private or public) mutable variables*](/common-programming-concepts/objects-and-classes/objects.html) could be declared stable:
+```motoko
+{{#include _mo/stable-var1.mo:c}}
+```
+
+*Immutable variable* `o1` and *mutable variable* `o2` can be stable because they have a value of an *object with private and public mutable variables*. 
+
+### Stable Objects with Private Functions
+Immutable or mutable variables of [*objects with private functions*](/common-programming-concepts/objects-and-classes/objects.html) could be declared stable:
+```motoko
+{{#include _mo/stable-var1.mo:d}}
+```
 
-The restriction for stable variables is shown in the last comment. We can NOT assign an object instance with mutable fields as the value of a stable variable, because an object with a mutable variable is not of shared type.    -->
+*Immutable variable* `p1` and *mutable variable* `p2` can be stable because they have a value of an *object with a private function*. 
 
+### Other stable types
+On top all [shared types](/internet-computer-programming-concepts/async-data/shared-types.html) and the types mentioned above, the following types could also be stable:  
+- [Tuples](/common-programming-concepts/types/tuples.html) of stable types 
+- [Option](/common-programming-concepts/options-and-results.html) types of any stable types
+- [Variant](/common-programming-concepts/types/variants.html) types with associated types that are stable
 
 ## How it works
-Declaring mutable variable(s) `stable` causes the following to happen *automatically* when [upgrading](/internet-computer-programming-concepts/basic-memory-persistence/upgrades.html) our canister with new actor code:
-- The value of the mutable variable(s) is *serialized* into [Candid format](/internet-computer-programming-concepts/async-data/candid.html#candid-serialization).
+Declaring variable(s) `stable` causes the following to happen *automatically* when [upgrading](/internet-computer-programming-concepts/basic-memory-persistence/upgrades.html) our canister with new actor code:
+- The value of the variable(s) is *serialized* into [Candid format](/internet-computer-programming-concepts/async-data/candid.html#candid-serialization).
```

Comment:
```suggestion
- The value of the variable(s) is *serialized* into [Candid format](/internet-computer-programming-concepts/async-data/candid.html#candid-serialization).
```
It's actually not Candid but an extension of Candid to support mutable data. I would just say serialized into an internal format, because we eventually plan to move away from this abuse of Candid. It's a long story...